sdk/azcore/runtime/policy_sas_credential.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package runtime
+
+import (
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/exported"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+// SASCredentialPolicy authorizes requests with a [azcore.SASCredential].
+type SASCredentialPolicy struct {
+	cred   *exported.SASCredential
+	header string
+}
+
+// SASCredentialPolicyOptions contains the optional values configuring [SASCredentialPolicy].
+type SASCredentialPolicyOptions struct {
+	// placeholder for future optional values
+}
+
+// NewSASCredentialPolicy creates a new instance of [SASCredentialPolicy].
+//   - cred is the [azcore.SASCredential] used to authenticate with the service
+//   - header is the name of the HTTP request header in which the shared access signature is placed
+//   - options contains optional configuration, pass nil to accept the default values
+func NewSASCredentialPolicy(cred *exported.SASCredential, header string, options *SASCredentialPolicyOptions) *SASCredentialPolicy {
```

Comment:
Hm...I don't think we want `exported` to show up in the public API surface since it's internal (can users create it?).

It might be that we need to put the credentials into a sibling package to runtime because what I've done below might be circular.

```suggestion
func NewSASCredentialPolicy(cred *azcore.SASCredential, header string, options *SASCredentialPolicyOptions) *SASCredentialPolicy {
```

Reply:
For better or worse, we follow the same pattern elsewhere (e.g. `NewBearerTokenPolicy()`) for this exact reason (i.e. avoid a circular dependency).  Also keep in mind that `exported.SASCredential` is type-aliased as `azcore.SASCredential` so it's publicly available (this is included in the doc comment).

I don't think we want to put these credentials into a separate package.  `TokenCredential` is in `azcore` and IMO we should keep them all together.

Reply:
I see. I have a QoL issue to discuss this at some point - it's not a great experience because your first auto-complete option in VSCode, for instance, is to the exported symbol. So you always have to fix it - in this case I think it's not so bad because you have to use a constructor + error and so you're never going to run into this particular symbol (hopefully).